### ISSUE TITLE
Proposal: ERC20 Token Creator on Kaia Blockchain

### ISSUE THREAD

**Issue by preetsinghmakkar** [kaiachain/kaia-dapp-mono#31]
Description
- This submission is an ERC20 token creation platform built with Next.js and ethers.js, designed specifically for the Kaia Blockchain. It provides a user-friendly web interface for creating customizable ERC20 tokens.

Key features include:
  - Mintable Tokens: Create tokens that can be minted by authorized addresses.
  - Burnable Tokens: Create tokens that can be burned, reducing the total supply.
  - Pausable Tokens: Create tokens that can be paused and unpaused, halting all transfers.
  - Capped Tokens: Create tokens with a maximum supply cap.
  - Time-Lock Tokens: Create tokens that are locked for a certain period.

The platform aims to simplify the token deployment process, offering flexibility and security for various token use cases on the Kaia Blockchain.

Bounty track
- Contribute a free to use DApp onto mainnet

**Comment by zxstim** [comment]
Approved. Make sure you let the token creator to be the token owner.

I think you can try to be modular as well so community can contribute more contract types to your token creation platform. Feel free to ping me for second opinion.

**Comment by preetsinghmakkar** [comment]
Hi @zxstim,

Thank you for approving my proposal and the valuable feedback. I wanted to let you know that I've submitted the pull request as per the guidelines, and I'm looking forward to your review.

I've ensured that the token creator is set as the token owner.

Preet Singh.

**Comment by zxstim** [comment]
Looks good. Approved for 500 KLAY bounty payout.

Upgrade the UI+UX and I will grant you another 500 KLAY bounty.

**Comment by zxstim** [comment]
Closing this issue. Open a new issue if you upgraded the UI+UX.

**Comment by preetsinghmakkar** [comment]
@zxstim Thanks for notifying me! I'll definitely submit my project to the Dorahacks showcase page for visibility. I'm also working on making my dApp more professional and polished, and I'll continue to enhance the token creator tool. I look forward to contributing more and earning additional bonuses!

**Comment by zxstim** [comment]
Couple suggestions for improvement:
- Better UI
- More types of tokens

Also, I encourage adopting your own brand/name to the bounty submission. I want you to own the product fully and we will support you.

**Comment by preetsinghmakkar** [comment]
Hey @zxstim, I have updated my Website. Review it.

What's New?

🔄 Rebranding: I have changed the name from kaia-TokenifyMe to TokenMaker to better reflect the sophisticated nature of our platform.

🎨 UI/UX Overhaul: We've completely revamped the user interface and user experience. The new design is more intuitive, sleek, and user-friendly, ensuring a smoother and more enjoyable experience when creating and managing your tokens.
 
 Website URL : https://tokenmaker-ruddy.vercel.app/
 
 Once you review it then i change the submission.md and i will submit it to the Hackathon.
